Changed the way resvs are submitted in the hour dst changes

arungrover opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
This test suite encounters failures when DST changes between the time reservation is submitted and the time reservation is supposed to end.

#### Describe Your Change
Problem is that when PTL submits this reservation it converts the end time epoch and that conversion sometimes results in end time being in the past.
Change is to detect that DST is going to change soon and submit reservations with duration instead of submitting with end time.
#### Link to Design Doc
NA

#### Attach Test and Valgrind Logs/Output
[test_resv.txt](https://github.com/openpbs/openpbs/files/5586691/test_resv.txt)



<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

bhroam commented on test/tests/functional/pbs_reservations.py:
```diff
@@ -1986,7 +1977,10 @@ def test_delete_idle_resv_job_standing(self):
+        delay = 3600
+        if self.dst_changes(start, start + delay) is True:
+            delay = 7200
+        start_str = time.strftime(strf_str, time.localtime(start + delay))
```
Is there a good reason we don't just use 7200 always?

bhroam commented on test/tests/functional/pbs_reservations.py:
```diff
@@ -1931,10 +1920,12 @@ def test_delete_idle_resv_job_delete(self):
+                                      select='1:ncpus=1',
+                                      start=start,
+                                      end=now + 3600,
+                                      extra_attrs=extra)
```
Why submit as TEST_USER, it is the default.  Unless you are checking for it to be TEST_USER, there is no need.  Here and other places.

arungrover replied:
This is most likely because of copy/paste. Since there is no harm in doing this, I'd like to leave it as is.

arungrover commented:
@bhroam thanks for reviewing the PR. I have addressed your comment by changing the freq of the recurrence rule to daily instead of hourly. Increasing the duration of the reservation was causing issues because the freq was hourly.

bhroam reviewed (approved):
LGTM

hirenvadalia commented:
closing and reopening to restart travis.

ashwathraop reviewed (approved):
LGTM
